app/lib/fetch-token.ts

```diff
@@ -2,8 +2,26 @@ type TokenResponse = {
   token: string;
 };
 
-export const fetchToken = async (apiKeyAuthentication = true) => {
-  return apiKeyAuthentication
-    ? 'xx697404a7-6cfd-48c6-93d1-30d73d17e07a' // demo api key
-    : ((await (await fetch('http://localhost:3000/token')).json()) as TokenResponse).token;
+export const fetchToken = async (request?: null | Request, apiKeyAuthentication = false) => {
```

Comment:
My two cents: Would be much clearer, and less `if/else` logic with default params that are hard to understand what they do except if you go read the code to have two distinct path/function, as opposed to one function that handle both path.

ie:

`useSearchToken()` -> generate search token, no if else, no boolean param
`useApiKey()` -> use hardcoded API key, no if else, no boolean param

And then when configuring engine:

```
export const engineConfig: CommerceEngineDefinitionOptions = {
  configuration: {
    accessToken: await (usePublicApiKey ? useApiKey(): useSearchToken()),
```